cmd/charm-ingest/main.go

```diff
@@ -93,6 +94,14 @@ func main() {
 		MaxDisk:       *maxDisk,
 		SoftDiskLimit: !*hardDiskLimit,
 	}
+	// Ensure we're logged into both stores before proceeding, so we
+	// won't get random interactions thrown up during the ingestion process.
+	if err := p.Src.Login(); err != nil && !isAdminError(err) {
```

Comment:
If you're only ingesting public charms then it isn't necessary to log in to the source charm store. It seems unreasonable to require it in that case.